src/main/scala/lms/core/codegen/c_codegen.scala

```diff
@@ -260,6 +260,14 @@ class ExtendedCCodeGen extends CompactCodeGen with ExtendedCodeGen {
   val libraryFlags = mutable.HashSet[String]()
   def registerLibrary(nLibraries: String*) = libraryFlags ++= nLibraries.toSet
 
+  val libraryPaths = mutable.HashSet[String]()
+  def registerLibraryPath(paths: String*) = libraryPaths ++= paths.toSet
+
+  def composePaths(paths: mutable.HashSet[String], option: String): String = {
```

Comment:
minor: `paths` can be a more general type, like `Iterable[String]`. I also would probably call this function `joinPaths`.